### ISSUE TITLE
About the steps to get the quantitative results of the paper

### ISSUE THREAD

**Issue by OctoberKat** [prajwalsingh/EEG2Image#2]
Hi, thanks for your kindly reply last time! 
I am trying to get the quantitative results (Inception Scores) of the paper, and the following are what I did:
1. Prepare environments and datasets;
2. Run "test_generation.py" -> generate "experiment/finalversion";
3. run "thoughtvis_inceptionscore.py" -> generate "experiments/thought_inceptionscore.txt"

Then, I can get the inception scores in the generated text file. However, the results are somehow different from the values in the paper. Inception scores I get are:

|  class  | 0  | 1  | 2  | 3  | 4  | 5  | 6  | 7  | 8  | 9  |
|  ----  | ----  |----  |----  |----  |----  |----  |----  |----  |----  |----  |
|  mean | 7.98  | 7.90 | 6.52 | 7.17 | 7.85 | 6.54 | 5.50 | 6.86 | 7.27 | 6.99 | 
| std | 0.55 | 0.27 | 0.56 | 0.46 | 0.52 | 0.51 | 0.53 | 0.53 | 0.33 | 0.39 |

The mean values and std values are both higher than reported. I want to know if there is a problem with my operations? Thank you!

**Comment by prajwalsingh** [comment]
Hi, 

You are getting the expected result, and the Inception score reported in the paper, considering the confidence interval and the variance of the generated images. Our initial plan is to report the FID score also, but we could not get the pre-trained model of ThoughtViz, so it does not seem like a fair comparison.

Thank you.

**Comment by YutLan** [comment]
Hi, thanks for your great work!
I am trying to get the quantitative results (Inception Scores) of the paper, and meets some problems.
The following were what I did:
1、Run "inference_inception.py" -> generate "experiments/inference_inception/211";
2、Run "inference_inceptionscore.py" -> generate the inception score for 5706 images;
But the results are only 1.1109942197799683, wha'ts the problem? 
Thank you!

**Comment by YutLan** [comment]
It seems the problem of inception score function? On is from inceptionv3 and the other one is another?

**Comment by prajwalsingh** [comment]
Hi, 

Thank you for going through our work. 

Apologies for the messy code structure.  I suggest generating the final set of images with test_generation code, not inference_inception.py. For calculating the inception score, you can use either inceptionscore.py or thoughtvis_inceptionscore.py. Yes, there are different versions of the inception score. To keep a fair comparison, we used the version as per the ThoughtViz paper. I will be removing the other inception score code.

**Comment by YutLan** [comment]
Thank you for your answers! I have reproduced the experiment results! Thanks!

**Comment by YutLan** [comment]
Hi, bro
     By the way. Have you ever tried to reproduce the results of the paper "Brain2Image" about image generation cited in your papers?  I try to reproduce their inception scores and meet some difficulties.

**Comment by prajwalsingh** [comment]
I did, but I was not getting the result as expected may be due to hyperparameters in GAN training.

**Comment by YutLan** [comment]
Thank you for your reply. I also did, the average inception score of my result is only 3.5. Maybe I should search the learning rates and seed.

**Comment by CoMo6** [comment]
Hello! can you release a version in pytorch?

**Comment by prajwalsingh** [comment]
Hi @FlyingCan, currently no such plan, but I have the Pytorch implementation of a similar GAN network. You can make modifications to it for the same [[link](https://github.com/prajwalsingh/MSQ-GAN-Pytorch)].

**Comment by NZW666666** [comment]
与上述OctoberKat一样我做了如下工作：
1. 准备环境和数据集;
2. 运行 “test_generation.py” -> 生成 “experiment/finalversion”;
3. 运行 “thoughtvis_inceptionscore.py” -> 生成 “experiments/thought_inceptionscore.txt”
我得到的 Inception 分数是：

	0	1	2	3	4	5	6	7	8	9
 	7.97	7.90	6.52	7.17	7.85	6.53	5.50	6.86	7.26	6.98
	0.55	0.26	0.56	0.45	0.52	0.51	0.52	0.53	0.33	0.38
 结果和OctoberKat类似，请问我要如何做才能得到论文中的初始分数呢，你上述的解释是说论文中用的是FID分数吗？ @prajwalsingh @OctoberKat @YutLan

**Comment by NZW666666** [comment]
> 谢谢你的回答！我已经复制了实验结果！谢谢！

@YutLan 同学你得到了论文中的inception score的均值和方差吗？

**Comment by NZW666666** [comment]
> 谢谢你的回答！我已经复制了实验结果！谢谢！

同学，你得到了论文中的初始分数的均值和方差吗 @YutLan
